src/main/java/com/openshift/internal/restclient/WatchClient.java

```diff
@@ -0,0 +1,199 @@
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.nio.IOControl;
+import org.apache.http.nio.client.methods.AsyncCharConsumer;
+import org.apache.http.nio.client.methods.HttpAsyncMethods;
+import org.apache.http.nio.protocol.HttpAsyncRequestProducer;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.openshift.internal.restclient.http.UnauthorizedException;
+import com.openshift.internal.restclient.model.KubernetesResource;
+import com.openshift.internal.restclient.model.Status;
+import com.openshift.restclient.IOpenShiftWatchListener;
+import com.openshift.restclient.IResourceFactory;
+import com.openshift.restclient.ISSLCertificateCallback;
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.authorization.IAuthorizationClient;
+import com.openshift.restclient.authorization.IAuthorizationStrategy;
+import com.openshift.restclient.authorization.ResourceForbiddenException;
+import com.openshift.restclient.http.IHttpStatusCodes;
+import com.openshift.restclient.model.IResource;
+
+/**
+  * Encapsulation of the logic to handle watching resources.
+  * @author jeff.cantrill
+  *
+  */
+public class WatchClient implements IHttpStatusCodes{
+	private static final Logger LOGGER = LoggerFactory.getLogger(WatchClient.class);
+	private URL baseUrl;
+	private Map<String, String> typeMappings;
+	private IResourceFactory factory;
+	private IAuthorizationStrategy strategy;
+	private IAuthorizationClient authClient;
+	private SSLContext sslContext;
+	private SSLCertificateCallbackAdapter sslCertAdapter;
+	
+	public WatchClient(URL baseUrl, Map<String, String> typeMappings, IResourceFactory factory, IAuthorizationStrategy strategy, IAuthorizationClient authClient, ISSLCertificateCallback certCallback) {
+		this.baseUrl = baseUrl;
+		this.typeMappings = typeMappings;
+		this.factory = factory;
+		this.strategy = strategy;
+		this.authClient = authClient;
+		sslCertAdapter = new SSLCertificateCallbackAdapter(certCallback);
+		try {
+			sslContext = SSLContexts.custom()
+					.loadTrustMaterial(null, sslCertAdapter).build();
+		} catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e) {
+			throw createOpenShiftException("Unable to initialize watch:",e);
+		}
+	}
+	
+	public void watch(String kind, String namespace, IOpenShiftWatchListener listener) {
+		try {
+			final String token = strategy.getToken();
+			final String resourceVersion = getResourceVersion(kind, namespace, token);
+			try(CloseableHttpAsyncClient httpClient = HttpAsyncClients
+					.custom()
+					.setSSLContext(sslContext)
+					.setHostnameVerifier(sslCertAdapter).build()){
+				final String endpoint = new URLBuilder(baseUrl, typeMappings)
+						.kind(kind)
+						.namespace(namespace)
+						.watch()
+						.addParmeter("resourceVersion", resourceVersion)
```

Comment:
typo in the method name addParmeter instead of addParameter. if it's a public method, then renaming it would be problematic for backward compat.